lib/avatar/avatar.tsx

```diff
@@ -1,89 +1,88 @@
-import _ from 'lodash';
 import React from 'react';
-
 import Badge from '../badge';
+import BEM, { BEMInterface } from '../utilities/ts/bem';
+import includes from '../utilities/ts/includes';
+import isUndefined from '../utilities/ts/is-undefined';
+import itemise from '../utilities/ts/itemise';
+import randomise from '../utilities/ts/randomise';
+
+interface BorderSizeInterface {
+  cx: string;
+  cy: string;
+  r: string;
+  viewBox: string;
+}
 
-import BEM from '../utilities/ts/bem';
-import namespace from '../utilities/ts/namespace';
+interface BorderSizesInterface {
+  l: BorderSizeInterface;
+  m: BorderSizeInterface;
+  s: BorderSizeInterface;
+}
 
-const COORDS = {
-  l: {
-    cx: `24`,
-    cy: `24`,
-    r: `23.5`,
-    viewBox: `0 0 48 48`
-  },
-  m: {
-    cx: `20`,
-    cy: `20`,
-    r: `19.5`,
-    viewBox: `0 0 40 40`
-  },
-  s: {
-    cx: `16`,
-    cy: `16`,
-    r: `15.5`,
-    viewBox: `0 0 32 32`
-  }
+const BORDER_SIZES: BorderSizesInterface = {
+  l: { cx: `24`, cy: `24`, r: `23.5`, viewBox: `0 0 48 48` },
+  m: { cx: `20`, cy: `20`, r: `19.5`, viewBox: `0 0 40 40` },
+  s: { cx: `16`, cy: `16`, r: `15.5`, viewBox: `0 0 32 32` }
 };
 
-type TSex = `female` | `male` | `undisclosed`;
+type SexTypes = `female` | `male` | `undisclosed`;
 
 interface Props {
   className?: string;
   firstName: string;
   href?: string;
   image?: string;
   modifiers?: string;
-  sex: TSex;
+  sex: SexTypes;
   style?: React.CSSProperties;
   value?: number;
 }
 
-const Avatar: React.FC<Props> = props => {
-  const Tag: keyof JSX.IntrinsicElements = _.isUndefined(props.href)
+const Avatar: React.FC<Props> = (props: Props) => {
+  const Tag: keyof JSX.IntrinsicElements = isUndefined(props.href)
     ? `div`
     : `a`;
 
-  const bem = BEM(`avatar`);
-  bem.addModifiers(props.modifiers);
-  bem.addClassNames(props.className);
+  const BEM_MODULE: BEMInterface = BEM(`avatar`);
+  const {
+    addClassNames,
+    addModifiers,
+    getElement,
+    getModifier,
+    getResult
+  }: BEMInterface = BEM_MODULE;
+
+  addModifiers(props.modifiers);
+  addClassNames(props.className);
 
   return (
     <Tag
-      className={bem.getResult()}
+      className={getResult()}
       style={props.style}
       href={props.href}
       title={props.firstName}>
-      {Object.keys(COORDS).map((size, index) => {
-        return (
-          _.includes(_.split(props.modifiers, ` `), size) &&
-          (!_.isUndefined(props.href) && (
+      {Object.keys(BORDER_SIZES).map(
```

Comment:
if youre not mapping anything to another array, forEach is the appropriate method

Reply:
I think there are 20 references to Array .map in this PR, probably all of them should be replaced by forEach

Reply:
I didnt check all of them but it is worth reviewing at least the most straightforward ones

Reply:
@DanMad disregard this one. It was just my lack of JSX understanding